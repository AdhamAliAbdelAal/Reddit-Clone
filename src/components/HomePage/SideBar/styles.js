import { styled } from '@mui/system';

export const SideBarContainer = styled('div')(({ theme }) => ({
  width: 312,
  backgroundColor: theme.palette.secondary.main,
  marginLeft: 24,
}));
