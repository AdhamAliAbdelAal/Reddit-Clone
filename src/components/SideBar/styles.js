import { styled } from '@mui/system';

export const SideBarContainer = styled('div')(() => ({
  width: 312,
  marginLeft: 24,
  position: 'relative',
}));
