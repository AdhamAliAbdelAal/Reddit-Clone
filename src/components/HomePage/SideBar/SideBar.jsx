import Communities from '../Communities/Communities';
import HomePageFooter from '../HomePageFooter/HomePageFooter';
import PersonalReddit from '../PersonalReddit/PersonalReddit';
import RedditPremium from '../RedditPremium/RedditPremium';
import { SideBarContainer } from './styles';

function SideBar() {
  return (
    <SideBarContainer>
      <Communities />
      <RedditPremium />
      <PersonalReddit />
      <HomePageFooter />
    </SideBarContainer>
  );
}

export default SideBar;
