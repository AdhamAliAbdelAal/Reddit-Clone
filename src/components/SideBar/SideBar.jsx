import { SideBarContainer } from './styles';
/**
 * This component works as a container for the right half components
 * like communities in home page
 *
 * @component SideBar
 * @property {React.Component} children -Components which will be wrapped in sidebar
 * @returns {React.Component} Container represents its children
 */

function SideBar({ children }) {
  return (
    <SideBarContainer>
      {children}
    </SideBarContainer>
  );
}

export default SideBar;
