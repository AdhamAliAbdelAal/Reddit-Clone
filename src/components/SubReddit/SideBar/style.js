import { styled } from '@mui/material/styles';

export const SideBarContainer = styled('div')(({ theme, createPost }) => ({
  [theme.breakpoints.between('0', '1000')]: {
    display: 'none',
  },
  width: 312,
  display: 'flex',
  flexDirection: 'column',
  marginLeft: 24,
  backgroundColor: 'transparent',
  position: 'relative',
  marginTop: createPost ? 70 : 16,
}));
export const CommunityContainer = styled('div')({
  background: 'white',
  borderRadius: 4,
  marginBottom: 16,
  maxWidth: '100%',
  backgroundColor: 'white',
});
