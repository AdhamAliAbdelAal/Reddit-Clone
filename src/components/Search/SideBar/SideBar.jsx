import { SideBarContainer, CommunityContainer } from './style';
import Community from './Community/Community';
import People from './People/People';
import PersonalReddit from '../PersonalReddit/PersonalReddit';
import BackHome from '../../BackHomeBottun/BackHome';

function SideBar(props) {
  const { subreddits, peoples } = props;
  return (
    <SideBarContainer>
      <CommunityContainer>
        <Community subreddits={subreddits} />
      </CommunityContainer>
      <CommunityContainer>
        <People peoples={peoples} />
      </CommunityContainer>
      <PersonalReddit />
      <BackHome />
    </SideBarContainer>
  );
}

export default SideBar;
