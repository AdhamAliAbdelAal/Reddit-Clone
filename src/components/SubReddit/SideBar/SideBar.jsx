import { SideBarContainer, CommunityContainer } from './style';

import BackHome from './BackHome/BackHome';
import About from './AboutSubReddit/About';
import Moderators from './Moderators/Moderators';
import Flirt from './Flirt/Flirt';

function SideBar() {
  return (
    <SideBarContainer>
      <CommunityContainer>
        <About />
      </CommunityContainer>
      <CommunityContainer>
        <Flirt />
      </CommunityContainer>
      <CommunityContainer sx={{ padding: '0px 12px' }}>
        <Moderators />
      </CommunityContainer>
      <BackHome sx={{ marginTop: -5 }} />
    </SideBarContainer>
  );
}

export default SideBar;
