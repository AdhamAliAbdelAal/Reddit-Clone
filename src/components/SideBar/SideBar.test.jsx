import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import SideBar from './SideBar';

describe(SideBar, () => {
  it('SideBar renders correctly', () => {
    const tree = renderer
      .create(
        <SideBar width={200}>
          <div>dummy</div>
        </SideBar>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
