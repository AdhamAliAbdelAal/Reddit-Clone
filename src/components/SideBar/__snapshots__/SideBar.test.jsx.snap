// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar SideBar renders correctly 1`] = `
<div
  className="css-q28llr"
>
  <div>
    dummy
  </div>
</div>
`;
